rust/gui-client/src-tauri/src/client/gui.rs

```diff
@@ -287,25 +278,26 @@ pub(crate) fn run(
 
                     // In a normal Rust application, Sentry's guard would flush on drop: https://docs.sentry.io/platforms/rust/configuration/draining/
                     // But due to a limit in `tao` we cannot return from the event loop and must call `std::process::exit` (or Tauri's wrapper), so we explicitly flush here.
+                    // TODO: This limit may not exist in Tauri v2
                     telemetry.stop();
 
                     tracing::info!(?exit_code);
                     app_handle.exit(exit_code);
+                    // This was unreachable in Tauri v1
+                    std::process::exit(exit_code);
```

Comment:
It would be great if we don't need to call this ourselves but can just return from `main`. If that requires more refactoring, can go into a follow-up!

Reply:
That's the funny thing, `App::exit` returns now, but `App::run` still doesn't! I'll comment it better